discord/ext/menus/pagination.py

```diff
@@ -0,0 +1,264 @@
+        ----------
+        button_name: :class:`str`
+            Name of the button to customize
+        button_label: :class:`str`
+            Label to display on the button
+        button_emoji:
+            Emoji to display on the button
+        button_style: :class:`~discord.ButtonStyle`
+            ButtonStyle to use for the button
+
+        Returns
+        -------
+        :class:`~PaginateButton`
+            The button that was customized
+        """
+
+        if button_name not in self.buttons.keys():
+            return False
+        button: PaginateButton = self.buttons[button_name]["object"]
+        button.label = button_label
+        button.emoji = button_emoji
+        button.style = button_style
+        return button
+
+    def update_buttons(self):
+        """Updates the display state of the buttons (disabled/hidden)"""
+        for key, button in self.buttons.items():
+            if key == "first":
+                if self.current_page <= 1:
+                    button["hidden"] = True
+                elif self.current_page >= 1:
+                    button["hidden"] = False
+            elif key == "last":
+                if self.current_page >= self.page_count - 1:
+                    button["hidden"] = True
+                if self.current_page < self.page_count - 1:
+                    button["hidden"] = False
+            elif key == "next":
+                if self.current_page == self.page_count:
+                    button["hidden"] = True
+                elif self.current_page < self.page_count:
+                    button["hidden"] = False
+            elif key == "prev":
+                if self.current_page <= 0:
+                    button["hidden"] = True
+                elif self.current_page >= 0:
+                    button["hidden"] = False
+        self.clear_items()
+        if self.show_indicator:
+            self.buttons["page_indicator"]["object"].label = f"{self.current_page + 1}/{self.page_count + 1}"
+        for key, button in self.buttons.items():
+            if key != "page_indicator":
+                if button["hidden"]:
+                    button["object"].disabled = True
+                    if self.show_disabled:
+                        self.add_item(button["object"])
+                else:
+                    button["object"].disabled = False
+                    self.add_item(button["object"])
+            elif self.show_indicator:
+                self.add_item(button["object"])
+
+        # We're done adding standard buttons, so we can now add any specified custom view items below them
+        # The bot developer should handle row assignments for their view before passing it to Paginator
+        if self.custom_view:
+            for item in self.custom_view.children:
+                self.add_item(item)
+
+    async def send(self, messageable: abc.Messageable, ephemeral: bool = False):
+        """Sends a message with the paginated items.
+        Parameters
```

Comment:
don't think this would be accepted by sphinx like that
```suggestion

        Parameters
```